appeals/web/src/server/appeals/appeal-details/lpa-questionaire/lpa-questionnaire.controller.js

```diff
@@ -15,35 +13,58 @@ import { objectContainsAllKeys } from '../../../lib/object-utilities.js';
 import { renderDecisionIncompleteConfirmationPage } from './outcome-incomplete/outcome-incomplete.controller.js';
 
 /**
- *
- * @param {string} appealId
- * @param {string} lpaQId
+ * @param {import('@pins/express/types/express.js').Request} request
  * @param {import("@pins/express/types/express.js").ValidationErrors | string | null} errors
  * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
- * @param {import('got').Got} apiClient
  */
-const renderLpaQuestionnaire = async (apiClient, appealId, lpaQId, response, errors = null) => {
-	const lpaQuestionnaireResponse = await getLpaQuestionnaireFromId(apiClient, appealId, lpaQId);
+const renderLpaQuestionnaire = async (request, response, errors = null) => {
+	let lpaQuestionnairePromise = lpaQuestionnaireService.getLpaQuestionnaireFromId(
+		request.apiClient,
+		request.params.appealId,
+		request.params.lpaQId
+	);
+	let appealDetailsPromise = appealDetailsService.getAppealDetailsFromId(
+		request.apiClient,
+		request.params.appealId
+	);
 
-	const mappedLpaQuestionnaireSections = mapLpaQuestionnaire(lpaQuestionnaireResponse);
-	const formattedSections = [];
-	for (const section of mappedLpaQuestionnaireSections) {
-		formattedSections.push(generateSummaryList(section));
-	}
+	const [lpaQuestionnaire, appealDetails] = await Promise.all([
+		lpaQuestionnairePromise,
+		appealDetailsPromise
+	]);
+	const session = request.session;
 
-	return response.render('appeals/appeal/lpa-questionnaire.njk', {
-		summaryList: { formattedSections },
-		errors
-	});
+	if (lpaQuestionnaire && appealDetails) {
+		const currentUrl = request.protocol + '://' + request.get('host') + request.originalUrl;
+		const pageComponents = lpaQuestionnairePage(
+			{ lpaQ: lpaQuestionnaire },
+			{ appeal: appealDetails },
+			currentUrl,
+			session
+		);
+		const shortAppealReference =
+			appealDetails?.appealReference.split('/')[
+				appealDetails?.appealReference.split('/').length - 1
+			];
```

Comment:
Perhaps the `appealShortReference` function (`appeals/web/src/server/lib/nunjucks-filters/appeals.js`) could be used here rather than duplicating the functionality?

Reply:
👍